templates/makefile-c++/makefile

```diff
@@ -0,0 +1,60 @@
+# Clean makefile based on https://stackoverflow.com/questions/30573481/how-to-write-a-makefile-with-separate-source-and-header-directories
+# Modified to use g++. 
+
+##############################
+#	USER DEFINED VARIABLES	 #
+##############################
+# Set the name of the target executable
+PROJECT_NAME := main
+# Add any arguments to the `make run` routine, if applicable 
+RUN_ARGS := 
+
+##############################
+#      STATIC VARIABLES      #
+##############################
+SRC_DIR := src
+OBJ_DIR := obj
+BIN_DIR := bin
+
+EXE := $(BIN_DIR)/$(PROJECT_NAME)
+SRC := $(wildcard $(SRC_DIR)/*.cpp)
+OBJ := $(SRC:$(SRC_DIR)/%.cpp=$(OBJ_DIR)/%.o)
+
+##############################
+#       COMPILER SETUP       #
+##############################
+CXX=g++
+CPPFLAGS := -Iinclude -MMD -MP
```

Comment:
Might make sense to break up this `-I` flag into a separate variable in the Makefile (Ex. https://github.com/ethanmpeterson/bare-metal/blob/master/STM32F446X/Makefile#L2). This way, users can easily can easily pull in headers from elsewhere in the repository. Can be handy when using / implementing libraries used in more than one project (ie CAL).